gadget/install/install.go

```diff
@@ -169,79 +242,35 @@ func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options
 		}
 		logger.Noticef("created new partition %v for structure %v (size %v) %s",
 			part.Node, part, part.Size.IECString(), roleFmt)
-		encrypt := (options.EncryptionType != secboot.EncryptionTypeNone)
-
 		if part.Role == gadget.SystemSave {
 			hasSavePartition = true
 		}
+		if part.Role != "" {
+			devicesForRoles[part.Role] = part.Node
+		}
 
-		if encrypt && roleNeedsEncryption(part.Role) {
-			var encryptionKey keys.EncryptionKey
-			var err error
-			timings.Run(perfTimings, fmt.Sprintf("make-key-set[%s]", roleOrLabelOrName(part)), fmt.Sprintf("Create encryption key set for %s", roleOrLabelOrName(part)), func(timings.Measurer) {
-				encryptionKey, err = keys.NewEncryptionKey()
-				if err != nil {
-					err = fmt.Errorf("cannot create encryption key: %v", err)
-				}
-			})
-			if err != nil {
-				return nil, err
+		// part.Node can be updated inside
```

Comment:
Unless I'm mistaken, we don't use the new part.Node() value, not in this function nor in the file below.

Reply:
I suspect it's used by the mount code below?

Reply:
this should probably mention that then (the mount op below)

Reply:
I've pushed changes that no longer require part.Node to be updated internally. Unfortunately the PR grew, but only a little bit.